simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/trafficLight/TrafficLightSet.java

```diff
@@ -10,7 +18,92 @@
 
 public class TrafficLightSet implements ISimulationAware {
     private ID id;
+    public List<TrafficLightInSet> TrafficLightSetList;
+    public List<TrafficLightInSet> InteriorListA;
+    public List<TrafficLightInSet> InteriorListB;
+    private static Logger_Interface LOG = Logger.getLoggerInstance(TrafficLightSet.class.getSimpleName());
+
+    /**
+     * Constructor
+     *
+     * @param id TrafficLightSet ID
+     */
+    public TrafficLightSet(ID id) {
+        this.id=id;
+        this.TrafficLightSetList =new ArrayList<TrafficLightInSet>();
+        this.InteriorListA =new ArrayList<TrafficLightInSet>();
+        this.InteriorListB =new ArrayList<TrafficLightInSet>();
+    }
+
+    /**
+     * Adds traffic lights to the List of traffic lights within one junction
+     *
+     * @param trafficLightLnSet object to be added to the list
+     */
+    public void addTrafficlight(TrafficLightInSet trafficLightLnSet, char type ){ // type: A or B, the type of the InteriorList
```

Comment:
I meant their state when talking about opposite. I see your point though, might be tricky to decide during the graph construction stage for the rules which ones should be in opposite states than the others.